erdjs/src/dapp/walletConnectProvider.ts

```diff
@@ -52,58 +52,26 @@ export class WalletConnectProvider extends EventTarget implements IDappProvider
      *
      */
     async login(): Promise<string> {
-        const chainId = WALLETCONNECT_ELROND_CHAIN_ID;
         let returnUrl = "";
         if (!this.walletConnector) {
             await this.init();
         }
+
         if (this.walletConnector?.connected) {
-            this.walletConnector.killSession();
+            await this.walletConnector.killSession();
             Logger.trace("WalletConnect login started but walletConnect not initialized");
-        } else {
-            await this.walletConnector?.createSession({ chainId }).then(() => {
-                const uri = this.walletConnector?.uri;
-                if (uri !== undefined) {
-                    returnUrl = uri;
-                }
-            });
+            return returnUrl;
         }
-        return returnUrl;
-    }
 
-    loginAccount = (address: string) => {
-        if (this.addressIsValid(address)) {
-            this.address = address;
-            this.dispatchEvent(this.onWalletConnectLogin);
-        } else {
-            if (this.walletConnector?.connected) {
-                this.walletConnector?.killSession();
+        await this.walletConnector?.createSession({ chainId: WALLETCONNECT_ELROND_CHAIN_ID }).then(() => {
```

Comment:
Sorry I missed this the first time, but this code can be a lot simpler:

```js
await this.walletConnector?.createSession({ chainId: WALLETCONNECT_ELROND_CHAIN_ID });
if (!this.walletConnector?.uri) {
    return "";
}

return this.walletConnector.uri; // This way you can also get rid of returnUrl variable, and return empty string in line 63

Reply:
done